drivers/soundwire/bus.c

```diff
@@ -999,13 +995,10 @@ int sdw_bus_prep_clk_stop(struct sdw_bus *bus)
 		slave_mode = slave->curr_clk_stop_mode;
 
 		if (slave_mode == SDW_CLK_STOP_MODE1) {
-			ret = sdw_slave_clk_stop_callback(slave,
-							  slave_mode,
-							  SDW_CLK_POST_PREPARE);
-
-			if (ret < 0) {
-				dev_err(&slave->dev,
-					"post-prepare failed:%d", ret);
+			ret = sdw_slave_clk_stop_callback(slave, slave_mode, SDW_CLK_POST_PREPARE);
```

Comment:
this one wasn't clear to me whether it can return `-ENODATA` and whether it should be considered a no-error too. Same below

Reply:
all PRE_PREPARE and POST_PREPARE callbacks can return -ENODATA if they initiate a register access and the device has gone off the rails. So yes we want to dismiss this case here.

Reply:
but I did change this function so that it doesn't print warnings inside, it makes it simpler.